src/Components/Admin/ObjectEditor/visualpaneview.js

```diff
@@ -0,0 +1,137 @@
+            name,
+            file_type,
+            object_type,
+            asset
+        ) => {
+            const response = await createPresignedLinkAndUploadS3(
+                { file_type: file_type, type: "image", file_content: asset },
+                handleError
+            );
+
+            const imagePrefix = process.env.REACT_APP_ADMIN_ASSET_PREFIX;
+            props.saveImage(caption, imagePrefix + response.data.s3_key);
+        };
+
+        const uploadImage = async () => {
+            handleUploadImageSubmit(name, type, "image", imageByteArray);
+            setUploaded(true);
+            setImageByteArray(null);
+        };
+
+        if (imageByteArray !== null) {
+            uploadImage();
+        }
+    }, [name, type, imageByteArray, caption, props, handleError]);
+
+    const handleUploadFileChange = async (event) => {
+        if (event.target.files && event.target.files[0]) {
+            const file = event.target.files[0];
+            await fileToByteArray(file, setImageByteArray);
+            setName(file.name);
+            setType(file.type.replace("image/", ""));
+        }
+    };
+
+    const addCaption = () => {
+        const newText = {
+            text: prompt("Enter the text for the puzzle: "),
+        };
+
+        if (newText.text) {
+            setCaption(newText.text);
+        }
+    };
+
+    const deleteCaption = () => {
+        setCaption("");
+    };
+
+    return (
+        <div>
+            <Typography component="div" variant="h5" className={classes.text}>
+                Upload Image {imageByteArray?.length}
+            </Typography>
+            <input
+                accept=".jpg,.jpeg,.png"
+                className={classes.input}
+                id="contained-button-file"
+                type="file"
+                onChange={handleUploadFileChange}
+            />
+            <label htmlFor="contained-button-file">
+                <Button
+                    className={classes.uploadButton}
+                    variant="contained"
+                    color="primary"
+                    component="span"
+                >
+                    Upload
+                </Button>
+            </label>
+            {uploaded ? <div>{name} successfully uploaded</div> : null}
+            {caption === "" ? (
+                <div>
+                    Add Caption
+                    <IconButton
+                        className={classes.addButton}
+                        aria-label="add"
+                        onClick={addCaption}
+                    >
+                        <AddIcon />
+                    </IconButton>
+                </div>
+            ) : null}
+            {caption !== "" ? (
+                <div>
+                    Caption: {caption}
+                    <IconButton onClick={() => deleteCaption()}>
+                        <DeleteForever />
+                    </IconButton>
+                </div>
+            ) : null}
```

Comment:
The conditions here are just negations of each other, so instead of null put the other div and this should still function the same way.